src/shared/ethtool-util.c

```diff
@@ -299,6 +299,13 @@ int ethtool_get_permanent_macaddr(int *ethtool_fd, const char *ifname, struct et
         return 0;
 }
 
+#define UPDATE(dest, val, flag) \
+        {                       \
+                typeof(val) _v = (val);                \
+                flag = flag || dest == _v;             \
+                dest = _v;                             \
+        }
```

Comment:
hard to grok what this does. can you add comment? and I think you mean dest != _v, not dest == _v, right?

Reply:
hmm, i think "flag" is a misnomer here, it's a boolean (a flag to me is always the thing that makes up a bitmask..) hence maybe rename flag → changed.

maybe avoid the double eval of flag so:

```c
…
{
        typeof(val) _v = (val);
        if (dest != _v)
                changed = true;
        dest = _v;
}
```

Reply:
maybe enclose the macro in `do {…} while(false)`

Reply:
> hard to grok what this does. can you add comment? and I think you mean dest != _v, not dest == _v, right?

Ouch!